src/hotspot/share/gc/g1/g1FullCollector.cpp

```diff
@@ -94,10 +94,14 @@ uint G1FullCollector::calc_active_workers() {
   uint current_active_workers = heap->workers()->active_workers();
   uint active_worker_limit = WorkerPolicy::calc_active_workers(max_worker_count, current_active_workers, 0);
 
+  // Finally consider the amount of used regions.
+  uint used_worker_limit = MAX2(heap->num_used_regions(), 1u);
```

Comment:
Should we actually end up here with zero used regions? It does not hurt, but seems superfluous

Reply:
No we should not, but I went with the super-safe option and added the `MAX2(...)`. If you like, I can remove it :)